### DISCUSSION TITLE
How do I change where a GitHub Action runs?

### DISCUSSION THREAD

**Discussion by rfalanga** [community/community#38360] [Other Feature Feedback, Questions, & Ideas] [answered]
Today I created my first GitHub Action using a template for ASP.NET Core. It performs a build on the main branch, which at this point is all I'm interested in doing. However, it failed with a "MSB1003: Specify a project or solution file" error message in the `dotnet restore` step. I'm certain it's an issue of the build not running in the directory where the .csproj file exists. So, I see two possible solutions. 

- `cd` into the folder where the .csproj file is
- direct the build to run where the .csproj file is located.

My question is which is most often done and how do I do it?

**Comment by airtower-luna** [comment, marked as answer]
Each `run` step gets a fresh shell, so `cd` only works if you do it within the same step.

The alternative is to [set `working-directory` for your `run` step](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun):

```yaml
- run: your command here
  working-directory: run/here/
```

You can also [set a default working directory](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_iddefaultsrun) for all `run` steps in the job.

**Reply by rfalanga** [reply]
Setting the default working directory did the trick. Thank you very much!

**Reply by ivansholokh** [reply]
yep, thanks. Here is sample code block:
```yml
jobs:
  build:
    runs-on: ubuntu-latest

    defaults:
      run:
        working-directory: {my-dir}
```
